packages/evm/contracts/test/MockInputValidatorPolicy.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity >=0.8.27;
+
+import { IEnclavePolicy } from "../interfaces/IEnclavePolicy.sol";
+import { AdvancedPolicy } from "@excubiae/contracts/policy/AdvancedPolicy.sol";
+import {
+    AdvancedChecker
+} from "@excubiae/contracts/checker/AdvancedChecker.sol";
+import { Check } from "@excubiae/contracts/interfaces/IAdvancedChecker.sol";
+
+/// @title BaseERC721Policy.
+/// @notice Policy enforcer for Enclave Input validation.
+/// @dev Extends BasePolicy with Enclave specific checks.
+contract MockInputValidatorPolicy is AdvancedPolicy, IEnclavePolicy {
+    error MainCalledTooManyTimes();
+
+    uint8 public inputLimit;
+    mapping(address subject => uint8 count) public enforced;
+
+    /// @notice Initializes the contract with appended bytes data for configuration.
+    /// @dev Decodes AdvancedChecker address and sets the owner.
+    function _initialize() internal virtual override {
+        bytes memory data = _getAppendedBytes();
+        (address sender, address advCheckerAddr, uint8 _inputLimit) = abi
+            .decode(data, (address, address, uint8));
+        _transferOwnership(sender);
+
+        ADVANCED_CHECKER = AdvancedChecker(advCheckerAddr);
+        SKIP_PRE = true;
+        SKIP_POST = true;
+        inputLimit = _inputLimit;
+    }
+
+    function enforceWithLimit(
```

Comment:
nit: maybe is better to maintain consistency in naming with `enforce` since we are using `super._enforce()`.

Reply:
This lead to errors as `AdvancedPolicy` which we are inheriting from doesn't have a virtual function.

Reply:
yeah, you just need to override the `_enforce()` without caring much. The `AdvancedPolicy` will take care of calling it from is `enforce()` external function! 

See [SemaphoreExcubiae](https://github.com/privacy-scaling-explorations/excubiae/blob/9bf739a43932051fa3aca4daac0b195deaad5d7b/packages/contracts/contracts/extensions/SemaphorePolicy.sol#L28) as example.

Reply:
This worked great thanks!